src/transform/let.js

```diff
@@ -44,6 +44,18 @@ export default function(ast, loggerInstance) {
             variableMarker.markModified(decl.id.name);
           }
         });
+
+
+        // Empty destructuring should be converted to const.
+        //
+        //     var [] = foo;
+        //     var {} = foo;
+        //
+        // Mix of empty destructuring and other variable declarations will be handled
+        // by other parts of the transform.
+        if (node.kind === 'var' && node.declarations.every(isEmptyDestructuring)) {
+          node.kind = 'const';
```

Comment:
Modifying the `node.kind` field here is definitely the **easy** way to fix this, but it doesn't align with how the rest of this transform is organized. Until now, there have been two distinct phases:

- first gather all the data of all the var-declarations to be modified.
- then go and convert the vars to let/const.

This change breaks this pattern. Now some var-declarations get modified during this first loop of gathering data, which breaks expectations of somebody reading this code in the future.

I'd rather leave these var-s unchanged than do it this way.

Reply:
At first, I was trying to do it in the transformation phase, but nothing about empty restructuring will be collected in the "gathering data" stage. It's _empty_. 😢  That's why I ended up putting it here.

Reply:
Yeah, it would need to involve some more reorganization. Currently this conversion loop goes through all variables, and takes the var-declarations of them, but as there are no variables here, so such empty declaration will never be reached.

It's definitely not an easy change to make. There's a currently a heavily beaked-in assumption that all variable declarations actually declare at least a single variable.

One possible middle-ground solution would be to separately gather these empty declaration nodes and then separately loop through them in the modification-phase. This way at least the modification and gathering phases will be kept separate, avoiding confusion in that area.

Reply:
Good idea. I will try to separate them.